plugins/modules/packaging/os/snap.py

```diff
@@ -227,59 +214,32 @@ def state_present(self):
                   "error output for more details.".format(cmd=self.vars.cmd)
         raise ModuleHelperException(msg=msg)
 
-    def state_absent(self):
-        self.validate_input_snaps()  # if snap doesnt exist, it will be absent by definition
-        actionable_snaps = [s for s in self.vars.name if self.is_snap_installed(s)]
+    def _generic_state_action(self, actionable_func, actionable_var, params=None):
+        actionable_snaps = [s for s in self.vars.name if actionable_func(s)]
         if not actionable_snaps:
             return
         self.changed = True
-        self.vars.snaps_removed = actionable_snaps
+        self.vars[actionable_var] = actionable_snaps
         if self.module.check_mode:
             return
-        params = ['classic', 'channel', 'state']  # get base cmd parts
+        if params is None:
+            params = ['state']
```

Comment:
Alternatively, have
```suggestion
            params = ['classic', 'channel', 'state']
```
here.